dali/operators/decoder/audio/audio_decoder_impl.cc

```diff
@@ -17,39 +17,62 @@
 
 namespace dali {
 
-template <typename T>
-span<char> as_raw_span(T *buffer, ptrdiff_t length) {
-  return make_span(reinterpret_cast<char*>(buffer), length*sizeof(T));
+std::pair<int64_t, int64_t> ProcessOffsetAndLength(const AudioMetadata &meta, float offset_sec, float length_sec) {
+  int64_t offset = 0;
+  int64_t length = meta.length;
+  if (offset_sec > 0.0f)
+    offset = static_cast<int64_t>(offset_sec * meta.sample_rate);
+  if (length_sec > 0.0f)
+    length = static_cast<int64_t>(length_sec * meta.sample_rate);
+
+  // Limit the duration to the bounds of the input
+  if ((offset + length) > meta.length) {
+    length = meta.length - offset;
```

Comment:
Are we okay with this silent correction of the length, or should we throw?

Reply:
It's a must - the length is in seconds, not in samples - and it's a float, so some rounding may occur.

Reply:
ok